authenticate/handlers.go

```diff
@@ -54,93 +59,80 @@ func (a *Authenticate) authenticate(w http.ResponseWriter, r *http.Request, sess
 	if session.RefreshPeriodExpired() {
 		session, err := a.provider.Refresh(r.Context(), session)
 		if err != nil {
-			return fmt.Errorf("authenticate: session refresh failed : %v", err)
+			return xerrors.Errorf("session refresh failed : %w", err)
 		}
-		err = a.sessionStore.SaveSession(w, r, session)
-		if err != nil {
-			return fmt.Errorf("authenticate: failed saving refreshed session : %v", err)
+		if err = a.sessionStore.SaveSession(w, r, session); err != nil {
+			return xerrors.Errorf("failed saving refreshed session : %w", err)
 		}
 	} else {
 		valid, err := a.provider.Validate(r.Context(), session.IDToken)
 		if err != nil || !valid {
-			return fmt.Errorf("authenticate: session valid: %v : %v", valid, err)
+			return xerrors.Errorf("session valid: %v : %w", valid, err)
 		}
 	}
 	return nil
 }
 
-// SignIn handles the sign_in endpoint. It attempts to authenticate the user,
-// and if the user is not authenticated, it renders a sign in page.
 func (a *Authenticate) SignIn(w http.ResponseWriter, r *http.Request) {
 	session, err := a.sessionStore.LoadSession(r)
 	if err != nil {
-		switch err {
```

Comment:
n.b.: If we get an error, we don't really care if it's expected or not. We want to log it, and repropmpt. Some spill over from just updating errors, but felt related enough.